docs/decisions/0015-serving-static-assets.rst

```diff
@@ -0,0 +1,125 @@
+
+The switching between the two would happen with a new Middleware class that is implemented in openedx-learning. We would use this Middleware to make sure that assets are never served through the Studio and LMS URLs, and that no other Studio or LMS views are served through the asset server URL. This will require new configuration settings.
+
+View Implementation
+~~~~~~~~~~~~~~~~~~~
+
+The Learning Core will implement a Django REST Framework APIView that will provide the core functionality for serving assets. This view will be subclassed and customized by apps in LMS and Studio that want to make use of this functionality.
+
+The Learning Core will also provide auth related endpoints.
+
+Permissions
+~~~~~~~~~~~
+
+The Learning Core will not implement any permissions checking. It will be the responsibility of an app in Studio or the LMS add permissions classes to their subclass of the APIView.
+
+For instance, Studio would likely implement a simple permissions checking model that only examines the learning context and restricts access to course staff. LMS might eventually use a much more sophisticated model that looks at the individual Component that an asset belongs to.
+
+Cookie Authentication
+~~~~~~~~~~~~~~~~~~~~~
+
+Authentication will use a domain cookie on the root assets domain (e.g. ``*.assets.sandbox.openedx.io``). The cookie will have the ``Secure`` and ``HttpOnly`` attributes.
+
+OPEN QUESTIONS:
+
+* What's the best way to do handoff between LMS/Studio and get that cookie information over to the asset domains?
+
+Masquerading
+~~~~~~~~~~~~
+
+We could theoretically add masquerading information to the cookie, but we would not implement in the first iteration.
+
+Performance at Scale
+~~~~~~~~~~~~~~~~~~~~
+
+The performance of this system for any particular asset view request should be similar to what already exists today. The difference in the long term is that by tying permissions to individual Components, we will in practice be caching far less than we do today, which will increase the load on the LMS.
+
+The high scale version of this approach will require having a CDN with programmable workers and a scalable object store that supports signed URLs (such as S3). The main objective would be to shift the file streaming burden out of Django and onto the CDN and object store.
+
+In Learning Core, we would implement an APIView (or possibly extend the asset-serving one) to return a JSON response of file metadata. This would include things like size, MIME type, last modified date, cache expiration policy, etc. The response would also contain a signed URL pointing to a object store resource, like S3. The CDN worker then does the fetch on that resource. We would create an example worker for at least CloudFlare.
```

Comment:
As mentioned above, I think we should just go with this approach right away, though for now we can substitute a minimal in-process python django app in lieu of an edge function.